### PR TITLE
Release/basket about us

### PR DESCRIPTION
**Pull request by mamont79**
1. Issue ticket number and link: [Sprint 4](https://dalhoupavel.atlassian.net/jira/software/projects/SCRUM/boards/1);
2. Screenshot:
image

3. Done 18.09.2023;
4. Description of changes:
  - [x] 4_01: Add "Add to Cart" Button on Product Cards
  - [x] 4_02: Integrate Catalog Page with Commercetools API
  - [x] 4_03: Implement Efficient Loading of Products on Catalog
  - [x] 4_07: Add Product to Cart Functionality for Non-Duplicated
  - [x] 4_10: Fetch and Display Cart Items Using commercetools API
  - [x] 4_11: Modify Quantity of Products in Cart
  - [x] 4_12: Implement "Remove from Cart" Button
  - [x] 4_14: Display Message for Empty Shopping Cart
  - [x] 4_15: Add Catalog Link to Empty Cart Message
  - [x] 4_18: Implement "Clear Shopping Cart" Button
  - [x] 4-19: Craft a Detailed Team Description with Photos
  - [x] 4_20: Incorporate and Link RS School Logo on the About Us
  - [x] 4_21: Implement Routing and Browser Navigation to Basket
  - [x] 4_22: Implement Routing and Browser Navigation to About US
  - [x] 4_23: Implement a Basket Icon in Header Navigation
  - [x] 4_24: Implement About Us Page Icon in Header Navigation
  - [x] 4_25: Add cart page
  - [x] 4_08: Remove Product from Cart Functionality
  - [x] 4_13: Recalculate Total Cost Upon Cart Updates
  - [ ] 4_16: Implement Promo Code Application Feature
  - [ ] 4_17: Display Original and Discounted Prices After Promo
5. Сomments:

### CODE REVIEW

File: src/pages/aboutUs/TeamMemeberCardsList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import TeamMemebersData from '../../../teamMemebersData.json';
+import { TeamMemberCard } from '../TeamMemberCard';
+import { StyledMemberCardsList } from './style';
+
+export function TeamMembersCardsList() {
+  return (
+    <StyledMemberCardsList>
+      {TeamMemebersData.map((member) => {
```
Comment (AnnaIgnatova) [inline src/pages/aboutUs/TeamMemeberCardsList/index.tsx:8]: don't start variable with uppercase

Comment (AnnaIgnatova) [inline src/pages/aboutUs/TeamMemeberCardsList/index.tsx:8]: use destructuring for member

File: src/pages/basket/cardsBlock/CartCard/components/TextBlock/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable no-console */
```
Comment (AnnaIgnatova) [inline src/pages/basket/cardsBlock/CartCard/components/TextBlock/index.tsx]: remove disable rules

File: src/pages/basket/cardsBlock/CartCard/components/TextBlock/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  StyledCartProductDescription,
+  StyledCartProductTitle,
+  StyledCartText,
+} from './style';
+import { ICartCardProps } from '../../../../types';
+
+export function TextBlock({ cardData }: ICartCardProps) {
+  const { title, attributes, quantity } = cardData || {};
+  console.log(title);
```
Comment (AnnaIgnatova) [inline src/pages/basket/cardsBlock/CartCard/components/TextBlock/index.tsx]: remove console.log

File: src/pages/basket/cardsBlock/CartCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  };
+
+  const handleDeleteCartItem = () => {
+    dispatch(deleteCartProduct(cartData));
+  };
+
+  return (
+    <StyledCartCardWrapper id={lineItemId}>
+      <StyledCardInfoWrapper>
+        <StyledImgWrapper style={{ backgroundImage: `url(${imgUrl})` }} />
```
Comment (AnnaIgnatova) [inline src/pages/basket/cardsBlock/CartCard/index.tsx:46]: why you use styled component and styles

Comment (AnnaIgnatova) [reply src/pages/basket/cardsBlock/CartCard/index.tsx:46]: you need to pass prop to styled compoentn

File: src/pages/basket/index.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import Order from './orderBlock';
+import { EmptyCart } from '../../components/emptyCart';
+
+export default function Basket() {
+  const dispatch = useAppDispatch();
+  const { cartFields } = useAppSelector((state) => state.cart);
+
+  useEffect(() => {
+    if (!cartFields) dispatch(fetchMeActiveCart());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (AnnaIgnatova) [inline src/pages/basket/index.tsx:15]: remove disable rules

File: src/pages/basket/orderBlock/index.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-console */
+/* eslint-disable no-unsafe-optional-chaining */
+/* eslint-disable no-param-reassign */
```
Comment (AnnaIgnatova) [inline src/pages/basket/orderBlock/index.tsx:3]: remove disable rules

File: src/pages/basket/orderBlock/index.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        accumulator += currentItem.productDiscountedPriceInCents;
+      }
+      return accumulator;
+    }, 0);
+  totalDiscountedPrice =
+    typeof totalDiscountedPrice === 'number' ? totalDiscountedPrice / 100 : '';
+
+  useEffect(() => {
+    if (!cartFields) dispatch(fetchMeActiveCart());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (AnnaIgnatova) [inline src/pages/basket/orderBlock/index.tsx:51]: remove disable rules

File: src/pages/basket/orderBlock/index.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  }, [cartFields]);
+
+  return (
+    <Formik
+      initialValues={{
+        promocode: '',
+      }}
+      onSubmit={(values) => {
```
Comment (AnnaIgnatova) [inline src/pages/basket/orderBlock/index.tsx:59]: replace function to separate func

File: src/pages/welcome/index.tsx [hunk 9]
```diff
@@ -1,26 +1,62 @@
+/* eslint-disable no-console */
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (AnnaIgnatova) [inline src/pages/welcome/index.tsx:1]: remove disable rules

File: src/pages/welcome/index.tsx [hunk 10]
```diff
@@ -1,26 +1,62 @@
...
+    } else {
+      getAnonimCartById();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (isAuth && !cartFields) dispatch(fetchMeActiveCart());
+  }, [cartFields]);
+
+  console.log(isAuth);
```
Comment (AnnaIgnatova) [inline src/pages/welcome/index.tsx]: console.og

**Review by AnnaIgnatova — CHANGES_REQUESTED** [review]

File: src/pages/aboutUs/TeamMemberCard/MemberCardBody/style.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import styled from 'styled-components';
+
+export const StyledMemberCardBodyWrapper = styled.div`
+  flex-grow: 1;
+
+  width: 50%;
+
```
[the remark this answers is not in the document]
Comment (m5production) [reply src/pages/aboutUs/TeamMemberCard/MemberCardBody/style.ts:7]: It looks more clear for me and, especially when there are many styles code, easier to read.

**Review by AnnaIgnatova — APPROVED** [review]

**Comment by AnnaIgnatova** [comment]
mission completed 🥳🥳

**Pull request merged**